### PR TITLE
Adjust Title component

### PR DESCRIPTION
**Pull request by bhumika-gupta**
Comments here: https://github.com/PuMProfessionals/pump-web-v2/pull/39#issuecomment-896020108

I adjusted some of the numbers for width and positioning of the BackArrow and TitleSection media queries of the Title component. I only adjusted it for the media query (mobile screen size), not a laptop/desktop screen size, because that's what I was told needed fixing, but I feel like the arrow looks a bit strange on the laptop/desktop screen size as well. I made sure to consult the Figma design for the mobile version but I think the title looks a bit different (goes onto next line), but overall still looks fine from what I can see

I've attached some screenshots of how it looks like on different screen sizes.

Screen Shot 2021-08-12 at 9 49 25 PM

Screen Shot 2021-08-12 at 9 50 04 PM

Screen Shot 2021-08-12 at 9 50 23 PM

Screen Shot 2021-08-12 at 9 50 55 PM

### CODE REVIEW

**Comment by xsharonhe** [comment]
Not really too sure about the specifications about this ticket since it was opened by Simon and Diana did the component so I'll let one of them review instead. Thanks for the quick PR!

**Comment by xsharonhe** [comment]
Just a thought and @noisya55616 or @hirundinidae  can confirm, but would it be easier to just on mobile put arrow directly on top? So that this is more a column? I think putting the arrow logo beside text on such a small width is risky.

**Comment by hirundinidae** [comment]
> Just a thought and @noisya55616 or @hirundinidae can confirm, but would it be easier to just on mobile put arrow directly on top? So that this is more a column? I think putting the arrow logo beside text on such a small width is risky.

I think we can try as a column. If not, the current position could be changed at smaller widths to be closer to the top of the text. (still feels like `back` arrow, no longer awkwardly above the text like my previous pr)

**Comment by bhumika-gupta** [comment]
> > Just a thought and @noisya55616 or @hirundinidae can confirm, but would it be easier to just on mobile put arrow directly on top? So that this is more a column? I think putting the arrow logo beside text on such a small width is risky.
> 
> I think we can try as a column. If not, the current position could be changed at smaller widths to be closer to the top of the text. (still feels like `back` arrow, no longer awkwardly above the text like my previous pr)

hmm i'm having a hard time visualizing this.
129293640-0030031b-9158-42db-bd87-8e3e5985da3c

 do you guys mean like this ^ ? 

129293640-0030031b-9158-42db-bd87-8e3e5985da3c-2

or this^ ?
feel free to download one of the images in my initial PR description and mark it up to show what you mean by "try as a column"
@xsharonhe @hirundinidae

**Comment by xsharonhe** [comment]
> > > Just a thought and @noisya55616 or @hirundinidae can confirm, but would it be easier to just on mobile put arrow directly on top? So that this is more a column? I think putting the arrow logo beside text on such a small width is risky.
> > 
> > 
> > I think we can try as a column. If not, the current position could be changed at smaller widths to be closer to the top of the text. (still feels like `back` arrow, no longer awkwardly above the text like my previous pr)
> 
> hmm i'm having a hard time visualizing this.
> 129293640-0030031b-9158-42db-bd87-8e3e5985da3c
> 
> do you guys mean like this ^ ?
> 
> 129293640-0030031b-9158-42db-bd87-8e3e5985da3c-2
> 
> or this^ ?
> feel free to download one of the images in my initial PR description and mark it up to show what you mean by "try as a column"
> @xsharonhe @hirundinidae

Second one would be easier. If no one reviews your PR by tmrw afternoon, I can review

**Comment by bhumika-gupta** [comment]
> > hmm i'm having a hard time visualizing this.
> > 129293640-0030031b-9158-42db-bd87-8e3e5985da3c
> > do you guys mean like this ^ ?
> > 129293640-0030031b-9158-42db-bd87-8e3e5985da3c-2
> > or this^ ?
> > feel free to download one of the images in my initial PR description and mark it up to show what you mean by "try as a column"
> > @xsharonhe @hirundinidae
> 
> Second one would be easier. If no one reviews your PR by tmrw afternoon, I can review

Okay, I'll try getting it done by tomorrow!

**Comment by bhumika-gupta** [comment]
Working on this was a bit weird since it shows up differently across different browsers (Safari and Chrome for example) and also my actual phone. I'll send screenshots of how it looks 

Safari:
Screen Shot 2021-08-16 at 12 09 34 AM

Chrome: 
Screen Shot 2021-08-16 at 12 10 37 AM

Screen Shot 2021-08-16 at 12 11 16 AM

Screen Shot 2021-08-16 at 12 11 36 AM

Screen Shot 2021-08-16 at 12 12 07 AM

On my actual phone:
IMG_8C0EA1E7A6C3-1

I feel like it being centered actually is more risky but anyways, I wasn't exactly sure how to set media queries for multiple breakpoints since when I set it for 600, only 900 applies.

**Comment by hirundinidae** [comment]
Sorry about the delay reviewing! The column looks ok at smaller widths, but I think it looks weird on a larger screen when the arrow can fit to the left of the text. (suggestions below) 
image

File: components/Title.jsx [hunk 1]
```diff
@@ -66,15 +66,17 @@ const STitle = styled.h1`
 
 const BackArrow = styled.div`
   position: absolute;
-  ${({ isLeftAligned }) => `
-    left: 80px;
-    top: ${isLeftAligned ? "90px" : "150px"};
-  `}
+  top: 100px;
+  // ${({ isLeftAligned }) => `
```
Comment (hirundinidae) [inline components/Title.jsx]: Remove if it's not used anymore? (don't keep commented out code)

**Comment by bhumika-gupta** [comment]
Is it better now? It has the same positioning for the normal screen as before (with the back arrow aligned on the left). On a screen less than 600px, it's in the middle still but I wasn't sure how to make so that it is not leftAligned for the media query so I just played around with numbers and put `left: 47%;`. If there's a better way to make it aligned in the center (which there probably is but I couldn't figure it out with my limited knowledge of React props), please let me know

**Comment by xsharonhe** [comment]
> Is it better now? It has the same positioning for the normal screen as before (with the back arrow aligned on the left). On a screen less than 600px, it's in the middle still but I wasn't sure how to make so that it is not leftAligned for the media query so I just played around with numbers and put `left: 47%;`. If there's a better way to make it aligned in the center (which there probably is but I couldn't figure it out with my limited knowledge of React props), please let me know

I still think this looks a bit weird - it's better though. This is my attempt at it:
```
import styled from "styled-components";
import PropTypes from "prop-types";
import Image from "next/image";
import Link from "next/link"; 

import LeftArrow from "../public/LeftArrow-NoCircle.svg";
import { baseTheme } from "../theme";
import { Text } from "../components";
import { media } from "../utils";

/**
 *
 * Title Component
 * @prop {string} title - Page title
 * @prop {string} arrowLink - Link of back arrow
 * @prop {string} description - Section description
 * @prop {string} image - Path to right-aligned image
 * @prop {number} imageWidth - width of right-aligned image, default 200px
 * @prop {number} imageHeight - heigh tof right-aligned image, default 200px
 * @prop {string} backgroundColor - Color of header, default navy (match navbar)
 * @prop {boolean} isLeftAligned - Is text left aligend
 * |- default true if image is present
 */

export const Title = ({
  title,
  arrowLink,
  description,
  image,
  imageWidth = 200,
  imageHeight = 200,
  backgroundColor = baseTheme.colors.navy,
  isLeftAligned = !!image,
  ...props
}) => (
  <Wrapper
    backgroundColor={backgroundColor}
    isLeftAligned={isLeftAligned}
    {...props}
  >
    <TopSection>
      {!!arrowLink && (
        <Link href={arrowLink}>
          <a>
            <Image src={LeftArrow} width={25} />
          </a>
        </Link>
      )}
      <TitleSection>
        <STitle>{title}</STitle>
        {!!description && <Text color="white">{description}</Text>}
      </TitleSection>
    </TopSection>
    {!!image && <Image src={image} width={imageWidth} height={imageHeight} />}
  </Wrapper>
);

const STitle = styled.h1`
  font-size: 3rem;
  color: white;
  font-weight: 900;
  display: flex;
  flex-direction: row;
  align-items: center;
  ${media(
    "tablet",
    `
      font-size: 2rem;
    `
  )}
`;
const TopSection = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 60%;
  ${media(
    "tablet",
    `
      flex-direction: column;
    `
  )}
`;

const TitleSection = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 5%;
  ${media(
    600,
    `
      margin-left: 0;
    `
  )}
`;

const Wrapper = styled.div`
  display: flex;
  /* TODO: replace with background svg */
  ${({ backgroundColor }) => `
    background-color: ${backgroundColor};
  `};
  ${({ isLeftAligned }) => `
    justify-content: ${isLeftAligned ? "space-between" : "center"};
    text-align: ${isLeftAligned ? "left" : "center"};
    padding: ${isLeftAligned ? "0 4rem 2rem" : "0 0 2rem"};
  `}
  ${media(
    "tablet",
    `
      flex-direction: column;
      justify-content: center;
      align-items: center;
      text-align: center;
      padding: 0 0 2rem;
    `
  )}
`;

Title.propTypes = {
  title: PropTypes.string,
  arrowLink: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.object,
  imageWidth: PropTypes.number,
  imageHeight: PropTypes.number,
  backgroundColor: PropTypes.string,
  isLeftAligned: PropTypes.bool,
};
```

I think it avoids having us to resort to ```px``` in absolute values

**Comment by hirundinidae** [comment]
Sharon's implementation makes sense (no absolute positioning), I would just increase the width of the TopSection by 10% and have the arrow more to the left of the screen
```
const TopSection = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 70%;  //instead of 60% 
  ```

**Comment by bhumika-gupta** [comment]
Ah I see what you did with the new div, which is also what you (or someone else - can't remember) were talking about initially. That makes a lot of sense! 

I personally think 60% width works better based on what I can see:
60% width:
Screen Shot 2021-08-19 at 7 36 52 PM

70% width: 
Screen Shot 2021-08-19 at 7 37 21 PM

It also looks quite good on my iPhone XR:
IMG_00032F4B4830-1

@xsharonhe @hirundinidae can you check again + confirm if it is okay?

File: components/Title.jsx [hunk 2]
```diff
@@ -38,19 +38,20 @@ export const Title = ({
     isLeftAligned={isLeftAligned}
     {...props}
   >
-    {!!arrowLink && (
-      <BackArrow isLeftAligned={isLeftAligned}>
+    {" "}
```
Comment (xsharonhe) [inline components/Title.jsx:41]: Confused why this is here but?

**Review by xsharonhe — APPROVED** [review]
Let's get this merged in for now :D

**Pull request merged**